### PR TITLE
fhc appforms forms export

### PR DESCRIPTION
**Pull request by corinnekrych**
@paolobueno here is my export feature wdyt?
@nialldonnellyfh would you kind a review?

### CODE REVIEW

**Comment by fh-automatron** [comment]
### Pull Request Quality Guidelines

As we continue to focus on quality, here are a few things to keep in mind:

* A component owner should review your change (we've tagged these above)
* This change should be deployed to a grid somewhere (via `fhcap-cli`) - please provide a link to this
* Automated testing has been added or updated for this change (e.g. UART, Unit/Acceptance tests)
* Ensure JIRA is up to date (e.g. deployment notes, tester notes, fix version, release notes)
* There should be an associated fhcap PR with this change - please provide a link to this
* Any SonarQube issues should be addressed where applicable (use your own judgement)

Please ping a component owner (see `fh contrib`) when your builds are complete, changes are deployed and you're ready for a review.

File: test/fixtures/appforms/fixture_forms.js [hunk 1]
```diff
@@ -27,4 +27,6 @@ module.exports = nock('https://apps.feedhenry.com')
   .delete('/api/v2/appforms/forms/someformid', '*')
   .reply(200, envReplies.crud)
   .post('/api/v2/appforms/forms/someformid/clone', '*')
+  .reply(200, envReplies.crud)
+  .get('/api/v2/appforms/forms/export', '*')
```
Comment (nialldonnellyfh) [inline test/fixtures/appforms/fixture_forms.js]: I think the reply from this should be a file.

File: test/unit/fh3/appforms/test_forms.js [hunk 2]
```diff
@@ -51,5 +54,17 @@ module.exports = {
       assert.equal(err, null);
       return cb();
     });
+  },
+  'test appforms-forms export': function(cb) {
+    appformsforms.export({ file: 'test/fixtures/appforms/fixture_export.zip'}, function (err) {
+      assert.equal(err, null);
+      fs.unlink('test/fixtures/appforms/fixture_export.zip', cb);
```
Comment (wei-lee) [inline test/unit/fh3/appforms/test_forms.js]: move the cleanup to the setup or tearDown functions to make sure it will be always executed.

Comment (nialldonnellyfh) [inline test/unit/fh3/appforms/test_forms.js]: Would be good to check that the file exists before unlinking.

**Comment by nialldonnellyfh** [comment]
Couple of comments above.

Overall looks good.

**Comment by wei-lee** [comment]
LGTM

File: test/unit/fh3/appforms/test_forms.js (hunk 2 above)
Comment (paolobueno) [inline test/unit/fh3/appforms/test_forms.js]: Maybe make this path the test zip to use with nock and instead write to os.tmpdir()

File: test/fixtures/appforms/fixture_forms.js (hunk 1 above)
Comment (paolobueno) [inline test/fixtures/appforms/fixture_forms.js]: We can have an empty file for now and add sample data from the backend endpoint later.

My test on the import endpoint is also wrong in this part, I'll change it to a 204 response with no content

File: lib/cmd/fh3/appforms/forms/export.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  },
+  'url': '/api/v2/appforms/forms/export',
+  'method': 'get',
+  'customCmd': function (params, cb) {
+
+    params.url = this.url;
+    params.method = this.method;
+    params.output = params.file;
+
+    if (params.file.indexOf(".zip") === -1) {
```
Comment (paolobueno) [inline lib/cmd/fh3/appforms/forms/export.js]: This is fragile, there's path.extname(). I think this fails for "some/my.zip/directory/", for instance

File: test/fixtures/appforms/fixture_forms.js (hunk 1 above)
Comment (corinnekrych) [inline test/fixtures/appforms/fixture_forms.js]: done

File: test/unit/fh3/appforms/test_forms.js (hunk 2 above)
Comment (corinnekrych) [inline test/unit/fh3/appforms/test_forms.js]: done

File: lib/cmd/fh3/appforms/forms/export.js (hunk 3 above)
Comment (corinnekrych) [inline lib/cmd/fh3/appforms/forms/export.js]: done

File: lib/cmd/fh3/appforms/forms/export.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    params.method = this.method;
+    params.output = params.file;
+
+    if (path.extname(params.file) !== ".zip") {
+      return cb("Expected the output file to have a .zip extension");
+    }
+
+    fs.stat(params.file, function (err) {
+      if (!err) {
+        return cb("The file at path " + params.file + " already exists.");
```
Comment (vchepeli) [inline lib/cmd/fh3/appforms/forms/export.js]: @corinnekrych this behavior is not tested in tests

**Comment by vchepeli** [comment]
LGTM

**Comment by paolobueno** [comment]
Obsoleted by #195

**Pull request closed without merging**